tmt/frameworks/shell.py

```diff
@@ -25,6 +25,8 @@ def get_test_command(
             logger: tmt.log.Logger) -> tmt.utils.ShellScript:
 
         # Use default options for shell tests
+        if guest.become:
+            return tmt.utils.ShellScript(f"{tmt.utils.SHELL_OPTIONS}; sudo {test.test}")
```

Comment:
I'm not sure this is the right spot. Or at least not the only one, because `beakerlib` framework should also honor `become`. `beakerlib` does not have its own implementation and inherits from framework base class, https://github.com/teemtee/tmt/blob/main/tmt/frameworks/__init__.py#L87.

Reply:
@happz , I updated the commit with a solution that does work. It works for connect and podman. I haven't had the chance to test it with beakerlib yet, but I'll do so as soon as I can.

Reply:
Actually, let me fix one more thing and I will ping when I have it ready. Thank you!

Reply:
What I'm currently looking into is this test case:

```
/whoami:
    summary: whoami
    test: "whoami && whoami"
```

The first `whoami` is `root`, the second is the user.

Reply:
I get everything to run and pass, including beakerlib, but I'm stuck on the framework inability to read the `TestResults` file:

```
        Read file '/var/tmp/tmt/run-100/plans/example/execute/data/guest/default-0/tests/smoketest/testbeakerlib-6/TestResults'.
        Unable to read '/var/tmp/tmt/run-100/plans/example/execute/data/guest/default-0/tests/smoketest/testbeakerlib-6/TestResults'.
```

Beside the regular tests, the beakerlib test itself passes from the beakerlib perspective, but not from the tmt perspective because of that file. The test:

```
#!/bin/bash
. /usr/share/beakerlib/beakerlib.sh || exit 1

rlJournalStart
    rlPhaseStartTest
        rlRun 'test "${USER}" == "root"' 0 "Good"
    rlPhaseEnd
rlJournalEnd
```

I suspect it is a permission issue, or a misconfiguration somewhere. I'm still looking into it.